src/cli.rs

```diff
@@ -221,9 +225,12 @@ pub fn run() -> Result<()> {
         root: conf_path,
     };
 
+    trace!("validate_config? {}", opts.validate_config);
     trace!("rekey? {}", opts.rekey);
     trace!("path.is_none()? {}", opts.path.is_none());
-    if opts.rekey && opts.path.is_none() {
+    if opts.validate_config {
+        validate_config(&conf)?;
```

Comment:
Minor nit, but I prefer to preface function calls with `self::` so it's obvious that the function is defined here, rather than imported from some other module.

```suggestion
        self::validate_config(&conf)?;
```